nat-lab/tests/test_direct_connection.py

```diff
@@ -39,265 +40,227 @@
+
+        print("Test direct -> relay -> direct transitions")
+        relay_events = await asyncio.gather(*[
+            exit_stack.enter_async_context(
+                run_async_context(
+                    client.wait_for_event_peer(
+                        node.public_key, [State.Connected], [PathType.Relay]
+                    )
+                )
             )
-            await exit_stack.enter_async_context(
-                beta_client.get_router().break_tcp_conn_to_host(str(server["ipv4"]))
+            for client, node in itertools.product(env.clients, env.nodes)
+            if not client.is_node(node)
+        ])
+
+        direct_events = await asyncio.gather(*[
+            exit_stack.enter_async_context(
+                run_async_context(
+                    client.wait_for_event_peer(
+                        node.public_key, [State.Connected], [PathType.Direct]
+                    )
+                )
             )
+            for client, node in itertools.product(env.clients, env.nodes)
+            if not client.is_node(node)
+        ])
 
-        await ping(alpha_connection, beta.ip_addresses[0])
+        async with _disable_direct_connection(env, reflexive_ips):
+            await asyncio.gather(*relay_events)
+            await _ping_between_all_nodes(env)
+
+        await asyncio.gather(*direct_events)
+        await _check_if_true_direct_connection(env)
+
+        print("Test direct connection on short connection loss")
+        possible_relay_events: Dict[Connection, Dict[Node, Any]] = defaultdict(dict)
+        for (client, conn), node in itertools.product(
+            zip(env.clients, env.connections), env.nodes
+        ):
+            if client.is_node(node):
+                continue
+            possible_relay_events[conn.connection][node] = (
+                await exit_stack.enter_async_context(
+                    run_async_context(
+                        client.wait_for_event_peer(
+                            node.public_key, [State.Connected], [PathType.Relay]
+                        )
+                    )
+                )
+            )
 
-        # LLT-5532: To be cleaned up...
-        alpha_client.allow_errors(
-            ["telio_proxy::proxy.*Unable to send. WG Address not available"]
-        )
-        beta_client.allow_errors(
-            ["telio_proxy::proxy.*Unable to send. WG Address not available"]
-        )
+        async with _disable_direct_connection(env, reflexive_ips):
+
+            async def ping_timeout(connection, node):
+                try:
+                    await ping(connection, node.ip_addresses[0], 10)
+                except asyncio.TimeoutError:
+                    pass
+                else:
+                    # if no timeout exception happens, this means, that peers connected through relay
+                    # faster than we expected, but if no relay event occurs, this means, that something
+                    # else was wrong, so we asserts
```

Comment:
I think random letter `s` appeared at the end of the comment :smile:, change `asserts` to `assert`